akka-persistence-testkit/src/main/scala/akka/persistence/testkit/scaladsl/UnpersistentBehavior.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.scaladsl
+
+import akka.actor.testkit.typed.scaladsl.BehaviorTestKit
+import akka.actor.typed.Behavior
+import akka.annotation.DoNotInherit
+import akka.persistence.testkit.internal.{ PersistenceProbeImpl, Unpersistent }
+
+import scala.reflect.ClassTag
+
+sealed trait UnpersistentBehavior[Command, State] {
+  val behavior: Behavior[Command]
+  lazy val behaviorTestKit = BehaviorTestKit(behavior)
+
+  def stateProbe: PersistenceProbe[State]
+}
+
+object UnpersistentBehavior {
```

Comment:
Looked into this and I see it would conflict with the existing EventSourcedBehaviorTestKit (which is somewhere in-between ActorTestKit async and BehaviorTestKit), so that would just be confusing.